File: 326-power-of-three/power-of-three.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n<=0:
            return False
        result=n//3
        if n==1:
            return True
        elif(n%3==0):
            return self.isPowerOfThree(result)
        return False
        

File: 326-power-of-three/power-of-three.cpp
class Solution {
public:
    bool isPowerOfThree(int n) {
        if (n <= 0) {
            return false;
        }
        int result = n / 3;
        if (n == 1) {
            return true;
        } else if (n % 3 == 0) {
            return isPowerOfThree(result);
        } else {
            return false;  
        }
    }
};
